### DISCUSSION TITLE
how to express the address range of 0x0 to 0xFFBF_FFFF in AddressSet?

### DISCUSSION THREAD

**Discussion by xlgforever** [chipsalliance/rocket-chip#3396] [Q&A]
I want to use the AddressSet to express the address range of 0x0 to 0xFFBF_FFFF, but it seems that there is no brief approach to do this. I must to write like:
```
address = Seq(AddressSet(0x00000000L, 0x7fffffffL),
    AddressSet(0x80000000L, 0xfffffffL),
    AddressSet(0x90000000L, 0xfffffffL),
    AddressSet(0xa0000000L, 0xfffffffL),
    AddressSet(0xb0000000L, 0xfffffffL),
    AddressSet(0xc0000000L, 0xfffffffL),
    AddressSet(0xd0000000L, 0xfffffffL),
    AddressSet(0xe0000000L, 0xfffffffL),
    ...)
```
Is there more elegant way to do this?

**Comment by ZenithalHourlyRate** [comment]
> Is there more elegant way to do this?

I'm afraid there is no elegant way.

Why such a range "0x0 to 0xFFBF_FFFF" is needed? Your final goal might be fullfilled in another way and such hacks might not be the right path.
